sftp-server.c

```diff
@@ -1706,14 +1706,16 @@ process_extended_home_directory(u_int32_t id)
 		fatal_fr(r, "parse");
 
 	debug3("request %u: home-directory \"%s\"", id, username);
-	if ((user_pw = getpwnam(username)) == NULL) {
+	if (username[0] != '\0') {
```

Comment:
I think it should be `username[0] == '\0'` as when the username is empty string, then we want the home directory for the current user `pw`

Reply:
Indeed. Thanks for the prompt comment! Fixed.